pageserver/src/tenant/timeline.rs

```diff
@@ -3530,6 +3553,47 @@ impl Timeline {
                     "keeping {} because it is the latest layer",
                     l.filename().file_name()
                 );
+                if l.is_incremental() {
+                    let mut to_remove: Vec<Key> = Vec::new();
+                    let mut insert_new_range = true;
+                    let start = l.get_key_range().start;
+                    let mut end = l.get_key_range().end;
+
+                    // ignore L0 layers
+                    if start != Key::MIN || end != Key::MAX {
```

Comment:
Could repurpose `!LayerMap::is_l0` here perhaps?
```suggestion
                    if !LayerMap::is_l0(&*l) {
```

But perhaps this definition is more broad than the `LayerMap::is_l0`; it most likely doesn't matter. Important to bound the number of l0 definitions.

Perhaps this could even be moved to the above `if` to get less indentation.